evm/vm/forks/frontier/headers.py

```diff
@@ -87,15 +87,16 @@ def create_frontier_header_from_parent(parent_header, **header_params):
 
 def configure_frontier_header(vm, **header_params):
     validate_header_params_for_configuration(header_params)
+    header = vm.block.header.as_mutable()
 
     for field_name, value in header_params.items():
-        setattr(vm.block.header, field_name, value)
+        setattr(header, field_name, value)
 
-    if 'timestamp' in header_params and vm.block.header.block_number > 0:
-        parent_header = get_parent_header(vm.block.header, vm.chaindb)
-        vm.block.header.difficulty = compute_frontier_difficulty(
+    if 'timestamp' in header_params and header.block_number > 0:
+        parent_header = get_parent_header(header, vm.chaindb)
+        header.difficulty = compute_frontier_difficulty(
             parent_header,
             header_params['timestamp'],
         )
 
-    return vm.block.header
+    return header
```

Comment:
This file's changes seem like a great example of the payoff to the approach: not risking a partial update to the header, in fact not even changing the current block header unless the caller decides to.